tests/infer/test_autoguide.py

```diff
@@ -11,16 +11,32 @@
 from pyro.infer import SVI, Trace_ELBO, TraceEnum_ELBO, TraceGraph_ELBO
 from pyro.infer.autoguide import (AutoCallable, AutoDelta, AutoDiagonalNormal, AutoDiscreteParallel, AutoGuideList,
                                   AutoIAFNormal, AutoLaplaceApproximation, AutoLowRankMultivariateNormal,
-                                  AutoMultivariateNormal, init_to_feasible, init_to_mean, init_to_median,
-                                  init_to_sample)
+                                  AutoMultivariateNormal, AutoProgressive, init_to_feasible, init_to_mean,
+                                  init_to_median, init_to_sample)
 from pyro.optim import Adam
 from tests.common import assert_close, assert_equal
 
 
+class AutoProgressive2(AutoProgressive):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.init_scale()
+
+
+class AutoProgressive3(AutoProgressive):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.init_scale()
+        self.init_cov()
```

Comment:
what's going on with `AutoProgressive2` and `AutoProgressive3`?

Reply:
They are test helpers to immediately skip to stages 2 and 3 respectively. I've added comments to that effect.